### ISSUE TITLE
Cannot run assembleAndroidTest in a library module that depends on stripe-terminal-1.0.8

### ISSUE THREAD

**Issue by isaac-udy** [stripe/stripe-terminal-android#99]
## Summary
The Stripe Terminal library includes a dependency on a .jar called `bbpossdk-1.7-all.jar`. This .jar includes JUnit classes that conflict with JUnit dependency in an androidTest variant, but only in a library module. 

Additionally it appears that the JUnit dependencies are a part of the main source set for bbpossdk, which means that JUnit testing annotations are available on the main source set for our project (which is not ideal, but less important than the fact we can't compile our UI tests). 

Running assembleAndroidTest on a library project results in many errors like this: 
  `Duplicate class org.junit.validator.ValidateWith found in modules jetified-junit-4.12.jar (junit:junit:4.12) and jetified-stripeterminal-1.0.8-runtime.jar (com.stripe:stripeterminal:1.0.8)`

And additionally has this issue: 
   `> More than one file was found with OS independent path 'junit/runner/logo.gif'`

## Code to reproduce
I have attached a .zip file containing a project that reproduces this issue. There are three modules in the project; app, library, and standaloneapp. 

If you run library:assembleAndroidTest, you will reproduce the issue. If you run standaloneapp:assembleAndroidTest you will see that an app module with a dependency on the stripe terminal SDK does not have the same problem. You can also see that running :app:assembleAndroidTest also succeeds, even though it has a dependency on the :library module.

## Android version
N/A - cannot compile androidTest build variant

## Impacted devices (Android devices or readers)
N/A - cannot compile androidTest build variant

## SDK version
1.0.8

## Other information
See attached .zip for sample project to reproduce

[bbposissues.zip](https://github.com/stripe/stripe-terminal-android/files/4264711/bbposissues.zip)

**Comment by chander-stripe** [comment]
Investigating now

**Comment by chander-stripe** [comment]
@isaac-udy this is definitely strange, and I'll investigate further tomorrow.

For now though, a quick fix to unblock you - 

You can add the following to the `build.gradle` file in the `library` module:

```
    configurations {
        compile.exclude group: "junit", module: "junit"
        test.exclude group: "junit", module: "junit"
        compile.exclude group: "org.hamcrest", module: "hamcrest-core"
        test.exclude group: "org.hamcrest", module: "hamcrest-core"
    }
```

right after the `defaultConfig` block.

After that, seems to work

```
➜  bbposissues ./gradlew :library:assembleAndroidTest

BUILD SUCCESSFUL in 3s
37 actionable tasks: 37 executed```

**Comment by isaac-udy** [comment]
Thanks @chander-stripe! The quick fix appears to work well for us.

**Comment by chander-stripe** [comment]
I've added the longterm fix for this, which will be released as part of version 1.0.15, to be released in the next 2 weeks.
Closing out for now, thanks for reporting!

**Comment by chander-stripe** [comment]
To be clear, once the fix on our side is in, folks won't have to add that `configurations` block I suggested above.
